File: tpa/desafiorecreio/python/recreio.py
#https://repl.it/languages/python3
entrada = open("entrada.txt", "r")
saida = open("saida.txt", "w")
n_turmas = int(entrada.readline())#lê a primeira linha, total de turmas
for linhas in range(n_turmas):
  n_alunos = int(entrada.readline())
  notas = entrada.readline()#Le a próxima linha com as notas
  chegada = notas.split()#Coloca as notas em cada aluno
  aluno = [] #Vetor que será reorganizado
  for ordem in chegada:
    aluno.append(ordem)
  print("Total de alunos = "+str(n_alunos))
  print("Chegada Aluno N")
  #Ordenar a lista em ordem decrescente de nota
  for i in range(n_alunos - 1):
    for j in range(i + 1,n_alunos):
      if(int(aluno[i]) < int(aluno[j])):
        aux = aluno[i]
        aluno[i] = aluno[j]
        aluno[j] = aux
  #Imprimir na tela a lista de chegada e ordenada
  for i in range(n_alunos):
    print(str(chegada[i])+" "+str(aluno[i])+" "+str(i+1))
  #Gravar no arquivo de saída o total de alunos que não trocaram
  naoTrocaram = 0;
  for i in range(n_alunos):
    if(chegada[i] == aluno[i]):
      naoTrocaram = naoTrocaram +1
  saida.write(str(naoTrocaram)+"\n")
#Fecha os dois arquivos
entrada.close()
saida.close()

File: tpa/desafiorecreio/php/recreio.php
<?php
	//Abrindo o arquivo de entrada
	if ($ponteiro = fopen("entrada.txt", "r")){
		//Percorrendo e mostrando o conteúdo na tela
		while(!feof($ponteiro)){
			$linha = fgets($ponteiro);
			echo $linha."<br/>";
		}
		fclose($ponteiro);
		//Abrindo novamente o arquivo de leitura e também o de gravação
		$entrada = fopen("entrada.txt", "r");
		$saida = fopen("saida.txt", "w");
		$nTurmas = intval(fgets($entrada));
		echo "Total de Turmas = ".$nTurmas."<br/>";
		for($x = 0; $x < $nTurmas; $x++){
			$nAlunos = intval(fgets($entrada));
			$arrayChegada = explode(" ",fgets($entrada));
			$arrayReordem = $arrayChegada;
			echo "Total de alunos = ".$nAlunos."<br/>";
			echo "Chegada Aluno N<br/>";
			//Ordenardenar a lista em ordem decrescente de nota
			for($i = 0; $i < $nAlunos - 1; $i++){
				for($j = $i + 1; $j < $nAlunos; $j++){
					if(intval($arrayReordem[$i]) < intval($arrayReordem[$j])){
						$aux = $arrayReordem[$i];
						$arrayReordem[$i] = $arrayReordem[$j];
						$arrayReordem[$j] = $aux;
					}
				}
			}
			//Imprimir na tela a lista de chegada e ordenada
			for($i = 0; $i < $nAlunos; $i++)
				echo "".$arrayChegada[$i]." ".$arrayReordem[$i]." ".$i."<br/>";
			//Gravar no arquivo de saída o total de alunos que não trocaram
			$naoTrocaram = 0;
			for($i = 0; $i < $nAlunos; $i++)
				if(intval($arrayChegada[$i]) == intval($arrayReordem[$i]))
					$naoTrocaram++;
			fwrite($saida, $naoTrocaram."\n");
		}
		//Fecha os dois arquivos
		fclose($saida);
		fclose($entrada);
	} else {
		echo "Erro ao abrir arquivo de leitura";
	}
?>